daemon/mgr/network.go

```diff
@@ -332,29 +332,61 @@ func (nm *NetworkManager) EndpointCreate(ctx context.Context, endpoint *types.En
 
 // EndpointRemove is used to remove network endpoint.
 func (nm *NetworkManager) EndpointRemove(ctx context.Context, endpoint *types.Endpoint) error {
+	var (
+		ep libnetwork.Endpoint
+	)
+
 	sid := endpoint.NetworkConfig.SandboxID
 	epConfig := endpoint.EndpointConfig
 
-	logrus.Debugf("remove endpoint: %s on network: %s", epConfig.EndpointID, endpoint.Name)
+	logrus.Debugf("remove endpoint(%s) on network(%s)", epConfig.EndpointID, endpoint.Name)
 
 	if sid == "" {
 		return nil
 	}
 
+	// find endpoint in network and delete it.
 	sb, err := nm.controller.SandboxByID(sid)
-	if err == nil {
-		if err := sb.Delete(); err != nil {
-			logrus.Errorf("failed to delete sandbox id: %s, err: %v", sid, err)
-			return err
+	if err != nil || sb == nil {
+		return errors.Errorf("failed to get sandbox by id(%s)", sid)
```

Comment:
I think we communicate before about how to deal with the err and nil.

First I think we should make sure the API of function `SandboxByID` to check if there is possibility of returning both a nil sandbox and nil-err; if it does not and returns an instance of sandbox and nil-err, we could simply judge `if err != nil`.

In the code 

```
if err != nil || sb == nil {
    return errors.Errorf("failed to get sandbox by id(%s)", sid)
}
```
It will miss the err message.